include/ddc/kernels/splines/spline_evaluator.hpp

```diff
@@ -557,27 +656,41 @@ class SplineEvaluator
                 return m_upper_extrap_rule(coord_eval_interest, spline_coef);
             }
         }
-        return eval_no_bc<eval_type>(coord_eval_interest, spline_coef);
+        return eval_no_bc(ddc::DiscreteElement<>(), coord_eval_interest, spline_coef);
     }
 
-    template <class EvalType, class Layout, class... CoordsDims>
+    template <class... DDims, class Layout, class... CoordsDims>
     KOKKOS_INLINE_FUNCTION double eval_no_bc(
+            ddc::DiscreteElement<DDims...> const& deriv_order,
             ddc::Coordinate<CoordsDims...> const& coord_eval,
             ddc::ChunkSpan<double const, spline_domain_type, Layout, memory_space> const
                     spline_coef) const
     {
+        using deriv_dim = ddc::Deriv<continuous_dimension_type>;
         static_assert(
-                std::is_same_v<EvalType, eval_type> || std::is_same_v<EvalType, eval_deriv_type>);
+                sizeof...(DDims) == 0
+                        || type_seq_same_v<detail::TypeSeq<DDims...>, detail::TypeSeq<deriv_dim>>,
+                "The only valid dimension for deriv_order is Deriv<Dim>");
+
+        auto const order = select_or(deriv_order, ddc::DiscreteElement<deriv_dim>(0)).uid();
+        KOKKOS_ASSERT(order >= 0 && order < 2)
+
         ddc::DiscreteElement<bsplines_type> jmin;
         std::array<double, bsplines_type::degree() + 1> vals_ptr;
         Kokkos::mdspan<double, Kokkos::extents<std::size_t, bsplines_type::degree() + 1>> const
                 vals(vals_ptr.data());
         ddc::Coordinate<continuous_dimension_type> const coord_eval_interest(coord_eval);
-        if constexpr (std::is_same_v<EvalType, eval_type>) {
+
+        if constexpr (sizeof...(DDims) == 0) {
             jmin = ddc::discrete_space<bsplines_type>().eval_basis(vals, coord_eval_interest);
-        } else if constexpr (std::is_same_v<EvalType, eval_deriv_type>) {
-            jmin = ddc::discrete_space<bsplines_type>().eval_deriv(vals, coord_eval_interest);
+        } else {
+            if (order == 0) {
```

Comment:
Can you move the definition of `order` in this `else` branch ? This would remove the call to `select_or`.